### PR TITLE
🚮Remove <amp-user-location>

### LINKED ISSUE #25573: Intent-to-Deprecate-and-Remove: <amp-user-location>
## Summary

`<amp-user-location>` uses geolocation API for precise user-location data, which supports the use case where dynamic data is tied to user location. See [Reference documentation](https://amp.dev/documentation/components/amp-user-location/).

Removal plan: Unregister the component, remove associated documentation and examples on amp.dev

## Motivation

No one is currently using it as it is only available under experimental. Additionally, the API is inconsistent with `<amp-geo>` which would be a more ideal component to eventually support the original use case anyway.

## Impact on existing users

None

## Alternative implementation suggestion for developers using AMP

We could eventually subsume the functionality of `amp-user-location` under `amp-geo` for relaunch with continued interest for the feature.

/cc @ampproject/wg-approvers @ampproject/wg-ui-and-a11y

### PR DESCRIPTION
**Pull request by caroqliu**
Closes #25573, #22177

Deletes `extensions/amp-user-location` and relevant materials in `examples/` or otherwise included via #22182

cc @ampproject/wg-caching I noticed there are references to `amp-user-location` in generated validator files--are these assumed to be overwritten via automated processes?

### CODE REVIEW

**Comment by honeybadgerdontcare** [comment]
@caroqliu if it's the generated protoascii, which puts all the protoascii files into one file, then it's fine. I only see three validator files with this and they're all being deleted in this PR. Since this is removing a component, have we verified it's not on currently valid pages?

**Comment by honeybadgerdontcare** [comment]
@caroqliu This is fine to be removed. It is not being used.

File: test/unit/test-url-replacements.js [hunk 1]
```diff
@@ -223,10 +223,10 @@ describes.sandboxed('UrlReplacements', {}, () => {
       const variables = Object.keys(
         replacements.getVariableSource().replacements_
       );
-      // Restrict the number of replacement params to globalVaraibleSource
+      // Restrict the number of replacement params to globalVariableSource
```
Comment (alanorozco) [inline test/unit/test-url-replacements.js]: Nice catch

**Comment by rcebulko** [comment]
Reviewer selection borked due to a bug, but that will no longer block it. The PR is still missing owners coverage however, and will probably require @jridgewell to approve (that's what the bot would suggest if it wasn't borking)

**Pull request merged**